# 오전장 vs 오후장
오전장: 09:00~12:00  
오후장: 12:00~15:30  
유튜브: https://youtu.be/VLNaFIUwPaA

In [6]:
import pandas as pd
import datetime
from scipy.stats import ttest_ind

In [7]:
df = pd.read_csv('data.csv')
df['매수시간'] = pd.to_datetime(df['매수시간'])
df['매수시간'] = df['매수시간'].apply(lambda x : x.time())
df['오전장'] = df['매수시간'] < datetime.time(12, 0, 0)
df.head()

주문일자     종목명  매수가격      매수시간      매도시간   수익률  매매순서 최대거래대금 시간  최대거래대금  \
0  2022-03-29   고려시멘트  5230  09:09:07  09:21:27  2.57     1  09:06:00     113   
1  2022-03-29   고려시멘트  5410  09:28:09  09:32:19  1.55     2  09:06:00     113   
2  2022-03-29    영진약품  5490  09:51:11  09:55:24  0.36     3  09:50:00      62   
3  2022-03-29  버킷스튜디오  5060  09:58:16  09:59:56 -0.89     4  09:22:00      72   
4  2022-03-29   우리바이오  4390  10:00:40  10:00:50 -0.87     5  10:00:00      44   

   직전 거래대금  양봉개수  10이격도  20이격도  60이격도      종목코드  매수등락률  시가등락률  뉴스기사   오전장  
0       63     1   4.38   9.72  13.32  198440.0  16.22   6.22     1  True  
1       16     2   0.93   1.72  11.55  198440.0  20.22   6.22     1  True  
2       62     1   1.57   1.94   4.43    3520.0   6.60   0.19     1  True  
3       26     0   0.87   2.06   4.17   66410.0  11.58   3.09     1  True  
4       43     3   3.56   4.06   4.23   82850.0  10.86   4.17     1  True

### Mean

In [8]:
df.groupby('오전장')['수익률'].mean()

오전장
False    0.150204
True    -0.015000
Name: 수익률, dtype: float64

### P-value

In [9]:
AM = df.loc[df['오전장'] == True].수익률
PM = df.loc[df['오전장'] == False].수익률

In [10]:
ttest_ind(AM, PM)

Ttest_indResult(statistic=-0.34711790576764456, pvalue=0.7290277346413371)

# Results
오전장 수익률 -0.015%    
오후장 수익률 0.15%  
(p-value = 0.729)

# Conclusion
오전장이 손실이고 오후장이 수익이다.   
또 한 번 직관에 반하는 충격적인 결과다.  
물론 통계적 유의성은 확보하지 못한 결과다.  
그럼에도 mean값이 예상과 반대라는 사실은 놀랍다.  